FreydCategoriesForCAP/gap/AdditiveClosure.gi

```diff
@@ -1356,6 +1362,256 @@ InstallGlobalFunction( INSTALL_FUNCTIONS_FOR_ADDITIVE_CLOSURE,
         
     fi;
     
+    if HasIsMonoidalCategory( underlying_category ) and IsMonoidalCategory( underlying_category ) then
+        
+        ##
+        AddTensorUnit( category,
+          function( cat )
+            
+            return ObjectConstructor( cat, [ TensorUnit ( UnderlyingCategory( cat ) ) ] );
+            
+          end );
+        
+        ##
+        AddTensorProductOnObjects( category,
+          function( cat, obj_1, obj_2 )
+            local underlying_category, objdat_1, objdat_2, len_1, len_2;
+            
+            underlying_category := UnderlyingCategory( cat );
+            objdat_1 := ObjectDatum( cat, obj_1 );
+            objdat_2 := ObjectDatum( cat, obj_2 );
+            len_1 := Length( objdat_1 );
+            len_2 := Length( objdat_2 );
+            
+            return ObjectConstructor( cat,
+                    List( [ 0 .. len_1 * len_2 - 1 ], i ->
+                      TensorProductOnObjects( underlying_category,
+                        objdat_1[1 + QuoInt( i, len_2 )],
+                        objdat_2[1 + RemInt( i, len_2 )] ) ) );
+            
+          end );
+        
+        ##
+        AddTensorProductOnMorphismsWithGivenTensorProducts( category,
+          function( cat, source, morph_1, morph_2, target )
+            local underlying_category, source_list, target_list, source_1, source_2, target_1, target_2,
+                  len_s1, len_s2, len_t1, len_t2, mat_1, mat_2;
+            
+            underlying_category := UnderlyingCategory( cat );
+            source_list := ObjectDatum( cat, source );
+            target_list := ObjectDatum( cat, target );
+            source_1 := Source( morph_1 );
+            source_2 := Source( morph_2 );
+            target_1 := Target( morph_1 );
+            target_2 := Target( morph_2 );
+            len_s1 := Length( ObjectDatum( cat, source_1 ) );
```

Comment:
The existing code in AdditiveClosure uses `ObjectList( object )` instead of `ObjectDatum( object )` (and `MorphismMatrix( morphism )` instead of `MorphismDatum( morphism )`) -> I suggest to keep this consistent.

Reply:
done